synapse/rest/media/v1/thumbnailer.py

```diff
@@ -30,6 +30,26 @@ class Thumbnailer(object):
 
     def __init__(self, input_path):
         self.image = Image.open(input_path)
+        try:
+            # We don't use ImageOps.exif_transpose since it crashes with big EXIF
+            image_exif = self.image._getexif()
+            image_orientation = image_exif.get(0x0112)
```

Comment:
could we use a symbolic constant here? https://pillow.readthedocs.io/en/3.0.x/reference/ExifTags.html

Reply:
The constants in PIL.ExifTags.TAGS are indexed reversed.
Looks like the dictionary is meant to display the values to the user instead of looking for them.

The value we want (`0x0112`) is the key if the dictionary, so this means we need to lookup by value (sub-optimal).

The value `0x0112` comes from EXIF standard, so its not PIL.ExifTags specific value that they can change.

In case we want to loopup by value, shoud we lookup for this key as a static code above instead of doing this each time?
Or maybe adding a comment in the same line would be enought?

Another option would be update to PIL 6 and try to fix ImageOps.exif_transpose for images with big EXIF.
The issue with the ImageOps.exif_transpose happends when they try to recreate the EXIF with no orientation tag after rotating, since we don't need EXIF at all this issue should not affect this project.

Reply:
Yeah, neither of those are good solutions. Fair enough, let's stick with the hardcoded number.